src/client/Microsoft.Identity.Client/TelemetryCore/Internal/Events/ApiEvent.cs

```diff
@@ -63,164 +35,102 @@ public enum ApiIds
         public ApiEvent(
             ICoreLogger logger,
             ICryptographyManager cryptographyManager,
-            string correlationId) : base(EventNamePrefix + "api_event", correlationId)
+            Guid correlationId)
         {
             _logger = logger;
-            _cryptographyManager = cryptographyManager;            
-        }
-
-        public ApiTelemetryId ApiTelemId
-        {
-            set => this[MsalTelemetryBlobEventNames.ApiTelemIdConstStrKey] = ((int) value).ToString(CultureInfo.InvariantCulture);
+            _cryptographyManager = cryptographyManager;
+            CorrelationId = correlationId;
         }
 
-        public ApiIds ApiId
-        {
-            get => TryGetValue(MsalTelemetryBlobEventNames.ApiIdConstStrKey, out string apiIdString) ? (ApiIds)Enum.Parse(typeof(ApiIds), apiIdString) : ApiIds.None;
+        public Guid CorrelationId { get; set; }
 
-            set => this[MsalTelemetryBlobEventNames.ApiIdConstStrKey] = ((int) value).ToString(CultureInfo.InvariantCulture);
-        }
+        public ApiIds ApiId { get; set; }
 
         public string ApiIdString
```

Comment:
nit: public string ApiIdString => ((int)ApiId).ToString(CultureInfo.InvariantCulture);

Reply:
Fixed in PR #3048.